File: C#/Circular Array.cs
// C# program to demonstrate use of circular
// array without using extra memory space

using System;
 

public class GfG {
 

    // function to print circular list

    // starting from given index ind.

    public static void print(char[] a, int n,

                                     int ind)

    {

        // print from ind-th index to

        // (n+i)th index.

        for (int i = ind; i < n + ind; i++)

            Console.Write(a[(i % n)] + " ");

    }
 

    // driver code

    public static void Main()

    {

        char[] a = new char[] { 'A', 'B', 'C',

                                'D', 'E', 'F' };

        int n = 6;

        print(a, n, 3);

    }
}
 
/* This code is contributed by vt_m */
 

File: Python/Circular Array.py
# Python3 program to demonstrate the use of 
# circular array without using extra memory space
 
# function to print circular list starting
# from given index ind.

def prints(a, n, ind):

    i = ind

     

    # print from ind-th index to (n+i)th index.

    while i < n + ind :

        print(a[(i % n)], end = " ")

        i = i + 1
 
# Driver Code

a = ['A', 'B', 'C', 'D', 'E', 'F']

n = len(a);

prints(a, n, 3);
 
# This code is contributed by rishabh_jain